### PR TITLE
Add --no-auth flag to disable authorization (e.g. in CI)

### LINKED ISSUE #172: Add --no-auth flag for `graph deploy`
Using keytar to store access tokens is nice, but it makes Graph CLI really hard to run in some environments, specifically headless Linux systems. It took me about 20 attempts to get the CLI to be able to access keytar -> libsecret -> D-Bus -> gnome-keyring in Travis, and that only worked in one of the three Ubuntu versions offered by Travis (trusty).

It would be great to make life a little easier for people who want to run Graph CLI in Docker for e.g. integration testing. A `--no-auth` flag could make `graph deploy` skip accessing keytar and therefor not need any of the above to work.

### PR DESCRIPTION
**Pull request by Jannis**
Resolves #172 by adding a `--no-auth` flag to `graph deploy`. This flag will override `--access-token` and will prevent accessing the system's keyring for previously saved access tokens.

### CODE REVIEW

File: graph.js [hunk 1]
```diff
@@ -228,26 +229,29 @@ app
...
-    } else {
-      try {
-        let node = normalizeNodeUrl(cmd.node)
-        accessToken = await keytar.getPassword('graphprotocol-auth', node)
-      } catch (e) {
-        logger.error(`Failed to fetch access token:`, e)
-        process.exitCode = 1
-        return
+    // If auth is enabled (default), add access token to the request
+    if (cmd.auth) {
```
Comment (orenyodfat) [inline graph.js]: does --no-auth flag automatically manifested at cmd.auth ? magic?

File: graph.js [hunk 2]
```diff
@@ -228,26 +229,29 @@ app
...
+          process.exitCode = 1
+          return
+        }
       }
-    }
 
-    // Use the access token, if one is sset
-    if (accessToken !== undefined && accessToken !== null) {
-      client.options.headers = { Authorization: 'Bearer ' + accessToken }
+      // Use the access token, if one is sset
```
Comment (leodasvacas) [inline graph.js]: Good chance to fix this typo: s/sset/set

**Review by leoyvens — APPROVED** [review]
lgtm after typo fix

**Comment by Jannis** [comment]
Obsoleted by #177.

**Pull request closed without merging**